qan/analyzer/mongo/profiler/aggregator/aggregator.go

```diff
@@ -49,51 +52,169 @@ type Aggregator struct {
 	// dependencies
 	config pc.QAN
 
-	// interval
-	timeStart time.Time
-	timeEnd   time.Time
-	D         time.Duration
-	stats     *stats.Stats
+	// status
+	status *status.Status
+	stats  *stats
+
+	// provides
+	reportChan chan *qan.Report
 
-	// make it safe to use from different threads
-	sync.Mutex
+	// interval
+	timeStart  time.Time
+	timeEnd    time.Time
+	d          time.Duration
+	t          *time.Timer
+	mongostats *mongostats.Stats
+
+	// state
+	sync.RWMutex                 // Lock() to protect internal consistency of the service
```

Comment:
Do we really need `Lock()`, etc methods to be a part of exported API? I don't think so.